custom_components/moonraker/number.py

```diff
@@ -80,6 +87,35 @@ async def async_setup_temperature_target(coordinator, entry, async_add_entities)
             )
             sensors.append(desc)
 
+        elif obj.startswith("temperature_fan"):
+            object_type, _, object_name = obj.partition(" ")
+            fan_name = object_name or object_type
+            fan_key = fan_name.replace(" ", "_")
+            display_name = fan_name.replace("_", " ").title()
+
+            settings = config_settings.get(obj, {})
```

Comment:
Moonraker normalises `configfile.settings` keys to lowercase, so names like `temperature_fan FAN_TEMP` end up stored as `"temperature_fan fan_temp"`. (Notice how we already call `obj.lower()` when we look up PWM pins just above.) Without a lowercase fallback here, any temperature fan whose name includes uppercase characters will ignore its configured `min_temp`/`max_temp`, fall back to 0–100, and we’re back to allowing out-of-range targets. Please mirror the existing pattern (try `config_settings.get(obj)` first, then `config_settings.get(obj.lower())`) so we always respect the user’s config.